src/classes/TDTM_Manager_TEST.cls

```diff
@@ -275,6 +275,178 @@ private class TDTM_Manager_TEST {
+
+		//get token again and verify the new handler is included
+		tokens = TDTM_Global_API.getTdtmConfig();
+        hasNewHandler = false;
+		for(TDTM_Global_API.TdtmToken token : tokens) {
+			if(token.targetObject == newHandlerToken.targetObject && token.className == newHandlerToken.className) {
+				hasNewHandler = true;
+			}
+		}
+		system.assertEquals(true,hasNewHandler);
+	}
+
+	//Test to make sure we can save Filter Field and Filter Value for default config
+	@isTest
+	public static void updateDefaultFilterFieldFilterValue() {
+		//get default tokens
+		List<TDTM_Global_API.TdtmToken> tokens = TDTM_Global_API.getDefaultTdtmConfig();
+
+		//Change the Filter Field and Filter Value
+		for(TDTM_Global_API.TdtmToken token : tokens) {
+			if(token.className == 'REL_Relationships_Con_TDTM') {
+				token.filterField = 'Gender__c';
+				token.filterValue = 'Male';
+			}
+		}
+
+		//use setTdtmConfig to save the filter in the DB
+		Test.startTest();
+		TDTM_Global_API.setTdtmConfig(tokens, 'hed');
+		Test.stopTest();
+
+		//verify the filter is saved in DB
+		List<Trigger_Handler__c> handlers = [select Id from Trigger_Handler__c where Class__c = 'REL_Relationships_Con_TDTM' and Filter_Field__c = 'Gender__c' and Filter_Value__c = 'Male' ];
+		system.assertEquals(1, handlers.size());
+
+	}
+
+	//Test to make sure we can save Filter Field and Filter Value for user's config
+	@isTest
+	public static void updateUserHandlerFilterFieldFilterValue() {
+
+		//Build Token for User's Trigger Handler
+		TDTM_Global_API.TdtmToken newHandlerToken = new TDTM_Global_API.TdtmToken('Contact_Update', 'Contact', 'BeforeInsert;BeforeUpdate', 0.2, true, false, null, null, false, '');
+
+		//Save the token into the DB
+		TDTM_Global_API.setTdtmConfig(new List<TDTM_Global_API.TdtmToken>{newHandlerToken});
+
+		//Verify that Filter is not in the DB
+		Trigger_Handler__c handler = [select Filter_Field__c, Filter_Value__c from Trigger_Handler__c where Class__c = 'Contact_Update'];
+		system.assertEquals(null, handler.Filter_Field__c);
+		system.assertEquals(null, handler.Filter_Value__c);
+
+		//Get token and add Filter Field and Filter Value
+		List<TDTM_Global_API.TdtmToken> tokens = TDTM_Global_API.getTdtmConfig();
+		for(TDTM_Global_API.TdtmToken token : tokens) {
+			if(token.className == 'Contact_Update') {
+				token.filterField = 'Gender__c';
+				token.filterValue = 'Male';
+			}
+		}
+
+		//set Filter field and Filter value
+		Test.startTest();
+		TDTM_Global_API.setTdtmConfig(tokens);
+		Test.stopTest();
+
+		//verify that the Filter is saved in DB
+		handler = [select Filter_Field__c, Filter_Value__c from Trigger_Handler__c where Class__c = 'Contact_Update'];
+		system.assertEquals('Gender__c', handler.Filter_Field__c);
+		system.assertEquals('Male', handler.Filter_Value__c);
+
+	}
+
+	//Test to make sure we will not overwrite Filter if it is marked as usermanaged
+	@isTest
+	public static void filterForUserManagedHandler() {
```

Comment:
Similar to the very first comment (the one on `STG_InstallScript_TEST`). Do we really need this test? It seems like the test `updatingDefaultHandlers` should already cover this scenario.  We could enhance that test instead of creating a new one (every new test adds to the time it takes to run our builds).

Reply:
I think you are correct, I can enhance updatingDefaultHandlers to include Filter stuff